File: common-client-libs/src/main/java/eu/getsoftware/hotelico/clients/infrastructure/dto/BasicDTO.java
package eu.getsoftware.hotelico.clients.infrastructure.dto;

import java.io.Serializable;

import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * Created by Eugen on 08.02.2016.
 */
@Getter
@NoArgsConstructor
public abstract class BasicDTO implements Serializable
{
	Long id = 0L;
	
	Long initId = 0L;
	
	Boolean active = true;
	
	String dtoType = this.getClass().getSimpleName().toLowerCase();
	
	BasicDTO(Long initId) {
        this.initId = initId;
    }
	
	public void setId(Long id)
	{
		this.id = id;
	}
	
	public void setInitId(Long initId)
	{
		this.initId = initId;
	}
}

File: hotel-client/src/main/kotlin/eu/getsoftware/hotelico/infrastructure/hotel/dto/BasicDTO.kt
package eu.getsoftware.hotelico.infrastructure.hotel.dto

import java.io.Serializable

/**
 * Created by Eugen on 08.02.2016.
 */
abstract class BasicDTO: Serializable
{
	var id : Long = 0L
	
	var initId : Long = 0L

	var active: Boolean = true

	val dtoType: String = this.javaClass.simpleName.toLowerCase()
}
